perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -379,68 +402,73 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
-      records_in_interval['start_time'],
-      records_in_interval['latency'],
-      records_in_interval['size'],
-      records_in_interval['stream_num'],
-      num_streams)
-  # A special throughput statistic that uses all the records, not
-  # restricted to the interval.
-  throughput_stats['net throughput (simplified)'] = (
-      records['size'].sum() * 8 / any_streams_active.duration
-      * units.bit / units.second)
-  gap_stats = analysis.GapStats(
-      records['start_time'],
-      records['latency'],
-      records['stream_num'],
-      all_streams_active,
-      num_streams)
-  logging.info('Benchmark overhead was %s percent of total benchmark time',
-               gap_stats['gap time proportion'].magnitude)
-
-  for name, value in itertools.chain(throughput_stats.iteritems(),
-                                     gap_stats.iteritems()):
-    results.append(sample.Sample(
-        'Multi-stream ' + operation + ' ' + name,
-        value.magnitude, str(value.units), metadata=distribution_metadata))
-
-  # QPS metrics
-  results.append(sample.Sample(
-      'Multi-stream ' + operation + ' QPS (any stream active)',
-      len(records) / any_streams_active.duration, '',
-      metadata=distribution_metadata))
-  results.append(sample.Sample(
-      'Multi-stream ' + operation + ' QPS (all streams active)',
-      len(records_in_interval) / all_streams_active.duration, '',
-      metadata=distribution_metadata))
-
   # Publish by-size and full-distribution stats even if there's only
   # one size in the distribution, because it simplifies postprocessing
   # of results.
-  for size in sizes:
-    this_size_records = records_in_interval[records_in_interval['size'] == size]
+  for size in all_sizes:
     this_size_metadata = metadata.copy()
     this_size_metadata['object_size_B'] = size
-    logging.info('Processing %s multi-stream %s results for object size %s',
-                 len(records), operation, size)
+    logging.info('Processing multi-stream %s results for object size %s',
+                 operation, size)
     _AppendPercentilesToResults(
         results,
-        this_size_records['latency'],
+        all_active_latencies[all_active_sizes == size],
         latency_prefix,
         LATENCY_UNIT,
         this_size_metadata)
 
+  # Throughput metrics
+  active_times = [np.sum(latency) for latency in active_latencies]
+  active_durations = [stop_times[i][active_stop_indexes[i] - 1] -
+                      start_times[i][active_start_indexes[i]]
+                      for i in xrange(num_streams)]
+  active_sizes = [np.sum(size) for size in active_sizes]
+  results.append(sample.Sample(
+      'Multi-stream ' + operation + ' net throughput',
+      np.sum((size / active_time * 8
+              for size, active_time in zip(active_sizes, active_times))),
+      'bit / second', metadata=distribution_metadata))
+  results.append(sample.Sample(
+      'Multi-stream ' + operation + ' net throughput (with gap)',
```

Comment:
I think it makes sense to report the throughput as seen once all workers are going;  if the time before that happens is so small as to be negligible (and you check that), then you can ignore the fact they don't all start at once as far as I'm concerned.  

But my bigger question here--even if you don't change the names, since you're using two names for the same thing, could you make sure to add a comment saying that?